jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocDriverUtils.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.dataproc.jdbc;
+
+import com.google.auto.value.AutoValue;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/** Helper methods to establish Connection for DataprocDriver. */
+public class DataprocDriverUtils {
+
+    static final String DEFAULT_DATABASE = "";
+    static final int DEFAULT_PORT = 10000;
+    static final String URL_HIVE_PREFIX = "jdbc:dataproc://hive/";
+
+    // Helper class to extract Hive connection parameters
+    @AutoValue
+    abstract static class HiveJdbcConnectionParam {
+        private static final String HIVE_PROTOCOL = "jdbc:hive2://";
+        private static final String TRANSPORT_MODE = "http";
+        private static final String HTTP_PATH = "cliservice";
+
+        // These client side params are required by DataprocDriver
+        abstract String host();
+
+        abstract String httpPath();
+
+        abstract String transportMode();
+
+        abstract int port();
+
+        abstract String dbName();
+
+        //  Do not parse these other parameters and pass them directly to HiveConnection
+        @Nullable
+        abstract String otherSessionConfs();
+
+        @Nullable
+        abstract String hiveConfs();
+
+        @Nullable
+        abstract String hiveVars();
+
+        static Builder builder() {
+            // Set default values
+            return new AutoValue_DataprocDriverUtils_HiveJdbcConnectionParam.Builder()
+                    .setHttpPath(HTTP_PATH)
+                    .setTransportMode(TRANSPORT_MODE)
+                    .setPort(DEFAULT_PORT)
+                    .setDbName(DEFAULT_DATABASE)
+                    .setOtherSessionConfs(null)
+                    .setHiveVars(null)
+                    .setHiveConfs(null);
+        }
+
+        public String toHiveJdbcUrl() {
+            String hiveJdbcURL =
+                    String.format(
+                            "%s%s:%d/%s;transportMode=%s;httpPath=%s",
+                            HIVE_PROTOCOL, host(), port(), dbName(), transportMode(), httpPath());
+            if (otherSessionConfs() != null)
```

Comment:
Apply google formatter

Reply:
This is the result from applying google-java-format-1.8, but I'll double check.

Reply:
I thought that google formatter will add `{` braces - it's not advised to use `if` statements without them, if it does not add them, then please add them manually to all `if` statements (even single-line).

Reply:
I see, thanks for mentioning that!